rllib/algorithms/impala/torch/vtrace_torch_v2.py

```diff
@@ -31,14 +31,16 @@ def make_time_major(
             for _tensor in tensor
         ]
 
-    assert (trajectory_len != recurrent_seq_len) and (
-        trajectory_len is None or recurrent_seq_len is None
+    assert (
+        trajectory_len is not None or recurrent_seq_len is not None
     ), "Either trajectory_len or recurrent_seq_len must be set."
 
     # Figure out the sizes of the final B and T axes.
-    if recurrent_seq_len:
-        B = recurrent_seq_len.shape[0]
-        T = tensor.shape[0] // B
+    if recurrent_seq_len is not None:
+        assert len(tensor.shape) == 2
+        # Swap B and T axes.
```

Comment:
As the customer asked today: is this conversion needed? Isn't it done by a connector? Our base LSTM encoder does not do it as far as I remember.

Reply:
It's a conversion specifically needed for v-trace, which sits inside the APPO/IMPALA loss. So I think it's a good design to make the loss itself do this flip (instead of the connector). Also, we only need to flip a few columns, not all of them.